concourse/steps/component_descriptor.py

```diff
@@ -77,6 +78,7 @@ def base_component_descriptor_v2(
               commit=commit,
             ),
             version=effective_version,
+            labels=source_labels,
```

Comment:
convert into a list. this on the one-hand-side will immediately validate that whathever is passed in is a least iterable, and also ensure that we always have a copy (thus avoiding unexpected side-effects from modifications done later-on).